### PR TITLE
qcachegrind: fix on Linux

### PR DESCRIPTION
**Pull request by JuPlutonic**
qwt.ru formula used as example for 26-38 lines (with addition of linux-g++)
cd do end block reduced, so now bin.install in linux is aslo working

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
Let's try to make determining the `spec` a bit more generic, and remove some of the lines we don't need. Also, it looks like we also need GCC on Linux:

```diff
diff --git a/Formula/qcachegrind.rb b/Formula/qcachegrind.rb
index e28ca623f1..badb5669d1 100644
--- a/Formula/qcachegrind.rb
+++ b/Formula/qcachegrind.rb
@@ -22,28 +22,32 @@ class Qcachegrind < Formula
   depends_on "graphviz"
   depends_on "qt@5"
 
+  on_linux do
+    depends_on "gcc"
+  end
+
+  fails_with gcc: "5"
+
   def install
     args = ["-config", "release", "-spec"]
-    spec = if OS.linux?
-      "linux-g++"
-    elsif ENV.compiler == :clang
-      "macx-clang"
+    os = OS.mac? ? "macx" : OS.kernel_name.downcase
+    compiler = case ENV.compiler
+    when :gcc
+      "g++"
     else
-      "macx-g++"
+      ENV.compiler
     end
-    spec << "-arm64" if Hardware::CPU.arm?
-    args << spec
+    arch = Hardware::CPU.intel? ? "" : "-#{Hardware::CPU.arch}"
+    args << "#{os}-#{compiler}#{arch}"
 
-    qt5 = Formula["qt@5"].opt_prefix
-    system "#{qt5}/bin/qmake", *args
+    system Formula["qt@5"].opt_bin/"qmake", *args
     system "make"
-    cd "qcachegrind" do
-      if OS.mac?
-        prefix.install "qcachegrind.app"
-        bin.install_symlink prefix/"qcachegrind.app/Contents/MacOS/qcachegrind"
-      else
-        bin.install "qcachegrind"
-      end
+
+    if OS.mac?
+      prefix.install "qcachegrind/qcachegrind.app"
+      bin.install_symlink prefix/"qcachegrind.app/Contents/MacOS/qcachegrind"
+    else
+      bin.install "qcachegrind/qcachegrind"
     end
   end
 end
```

**Comment by JuPlutonic** [comment]
linux-g++ needs to be intetrlaced to linux-gcc-11 inside qcb.pro file

File: Formula/qcachegrind.rb [hunk 1]
```diff
@@ -22,20 +22,32 @@ class Qcachegrind < Formula
...
-      if OS.mac?
-        prefix.install "qcachegrind.app"
-        bin.install_symlink prefix/"qcachegrind.app/Contents/MacOS/qcachegrind"
-      else
-        bin.install "qcachegrind/qcachegrind"
-      end
+    args = ["-config", "release", "-spec"]
+    os = OS.mac? ? "macx" : OS.kernel_name.downcase
+    compiler = case ENV.compiler.to_s.split("-").first
+    when :gcc
```
Comment (carlocab) [inline Formula/qcachegrind.rb:35]:
```suggestion
    compiler = if ENV.compiler.to_s.start_with?("gcc")
```
This is probably simpler.

Comment (JuPlutonic) [reply Formula/qcachegrind.rb:35]: ok

**Review by carlocab — APPROVED** [review]
Thanks @JuPlutonic!

For future reference, our preferred commit message format here would be something like
```
qcachegrind: fix on Linux
```
In particular, we always put formula names first in the commit message.

**Pull request merged**